### PR TITLE
Fixes#27314 - Handled issue in parsing "exit" in dellos6 running config 

### LINKED ISSUE #27314: Issue in handling of "exit" in parsing dellos6 running config
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
ansible.module_utils.dellos6

##### ANSIBLE VERSION
```
ansible 2.3.2.0 (stable-2.3 9d435fd47b) 
config file = /etc/ansible/ansible.cfg
configured module search path = Default w/o overrides


```

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```

```

### PR DESCRIPTION
**Pull request by abirami-n**
##### SUMMARY

Included changes to handle "exit" in dellos6 running-config.
Fixes #27314

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

ansible.module_utils.dellos6

##### ANSIBLE VERSION

```
ansible 2.4.0  (devel b4b3ca9d64) 
config file = /etc/ansible/ansible.cfg 
configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']

```

##### ADDITIONAL INFORMATION

Included change in plugin file of dellos6.py to support in devel.

### CODE REVIEW

**Comment by abirami-n** [comment]
The module_utils changes has to be back-ported to 2.3 and 2.2.

**Comment by gundalow** [comment]
@skg-net Could you please review this.

This will be for `devel` and `stable-2.3`.

`stable-2.2` will only get critical security bug fixes.

**Review by skg-net — APPROVED** [review]

**Comment by gundalow** [comment]
rebuild_merge

**Comment by gundalow** [comment]
Given that we are close to 2.4, this is for `devel` and `stable-2.4` only

File: lib/ansible/plugins/terminal/dellos6.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def on_authorize(self, passwd=None):
 
         cmd = {u'command': u'enable'}
         if passwd:
-            cmd[u'prompt'] = to_text(r"[\r\n]?password: $", errors='surrogate_or_strict')
+            cmd[u'prompt'] = to_text(r"[\r\n]?password:$", errors='surrogate_or_strict')
```
Comment (abadger) [inline lib/ansible/plugins/terminal/dellos6.py:51]: Prexisting but note that you're still operating on a string literal here so you can write it like this:
``` python
cmd[u'promt'] = ur"[\r\n]?password:$"
```

Comment (abadger) [reply lib/ansible/plugins/terminal/dellos6.py:51]: string literals are quicker than calling to_text() so that's why I mention it.

**Comment by abadger** [comment]
cherry-picked to stable-2.4 for the 2.4.0rc3 release.

**Comment by skg-net** [comment]
Thanks @abadger

**Pull request merged**